File: Flood_Fill/Flood_Fill.go
package main

import "fmt"

func main() {
	image := [][]int{{1, 1, 1}, {1, 1, 0}, {1, 0, 1}}
	sr := 1
	sc := 1
	color := 2
	fmt.Println(floodFill(image, sr, sc, color))
}

func floodFill(image [][]int, sr, sc, color int) [][]int {
	c := image[sr][sc]
	floodFillUitls(&image, sr, sc, c, color)
	return image
}

func floodFillUitls(image *[][]int, sr, sc, c, color int) {
	if sr < 0 || sr >= len(*image) {
		return
	}
	if sc < 0 || sc >= len((*image)[0]) {
		return
	}
	if (*image)[sr][sc] == color {
		return
	}
	if (*image)[sr][sc] != c {
		return
	}
	(*image)[sr][sc] = color

	floodFillUitls(image, sr+1, sc, c, color)
	floodFillUitls(image, sr-1, sc, c, color)
	floodFillUitls(image, sr, sc+1, c, color)
	floodFillUitls(image, sr, sc-1, c, color)

}

File: Flood_Fill/Flood_Fill.py
def floodFill(image,sr,sc,color):
    n = len(image)
    m = len(image[0])
    c = image[sr][sc] = color

    def floodFillUtils(image,n,m,sr,sc,color):
        if sr < 0 or sr >= n:
            return
        if sc < 0 or sc >= m:
            return

        if image[sr][sc] == color:
            return
        
        if image[sr][sc] != c:
            return

        image[sr][sc]=color
        
        floodFillUtils(image,n,m,sr-1,sc,color)
        floodFillUtils(image,n,m,sr+1,sc,color)
        floodFillUtils(image,n,m,sr,sc-1,color)
        floodFillUtils(image,n,m,sr,sc+1,color)

    floodFillUtils(image,n,m,sr,sc,color)
    return image

 





image = [[1,1,1],[1,1,0],[1,0,1]]
sr = 1
sc = 1
color = 2
print(floodFill(image,sr,sc,color))